### PR TITLE
Jfshaw1/feature/confirm delete menu

### PR DESCRIPTION
**Pull request by JoshuaShaw-13**
Added confirm deletion menus for deleting items in list view and detail view and also for deleting tags. Let me know if there are other places we delete things that require the menu.

Problem with tags: 
 - When selecting tags there is no indication that they are selected in UI.
 - When delete is pressed, tags that were selected go away but come back after fragment is reloaded

Note: these problems were present before I had made my changes, I am not sure if they are known problems

### CODE REVIEW

**Comment by lizard-52** [comment]
At least the first of those problems is know an is fixed (?) in Ben's new UI branch. Not sure about the other.

**Comment by lizard-52** [comment]
Second issue is fixed in my tag colour PR

**Comment by JoshuaShaw-13** [comment]
Previous issues seem to be resolved now, but app crashes when adding a new tag and when deleting multiple tags (but not when deleting a single tag). After restarting the app when it crashes from adding a tag the tag is actually created and appears in the tag manager. But after restarting from a crash because of deleting multiple tags the tags are not deleted.

**Comment by lizard-52** [comment]
I'm not worried about those bugs, we're doing an overhaul of the tag manager anyways (in ben's UI PR).

**Comment by JoshuaShaw-13** [comment]
Sounds good, then I think this PR is good to review

**Review by waridh — APPROVED** [review]
Cool, it works. Thank you.

**Review by lizard-52 — CHANGES_REQUESTED** [review]
Please factor the menu out into a method somewhere. We will need it in more places

File: code/app/src/main/java/com/example/sigma_blue/activities/ViewListActivity.java [hunk 1]
```diff
@@ -176,8 +178,27 @@ private void setUIOnClickListeners() {
 
         viewHolder.optionsButton.setOnClickListener(v -> {});
 
-        viewHolder.deleteSelectedButton.setOnClickListener(v ->
-            this.deleteSelectedItems());
+        viewHolder.deleteSelectedButton.setOnClickListener(v -> {
```
[the remark this answers is not in the document]
Comment (waridh) [reply code/app/src/main/java/com/example/sigma_blue/activities/ViewListActivity.java:184]: If its going to be reused, Nikolai is right. I don't know this side of the application, so I am not sure how much reuse we are planning out of it.

**Comment by JoshuaShaw-13** [comment]
Method has been made into an interface. I don't know if this is the best way to implement it but I wanted to make the change to get feedback if any (that's why its in placeholder). I can change it up tomorrow if needed, but wanted to get this out before stopping for the day

**Pull request merged**